cg/apps/demultiplex/sample_sheet/api.py

```diff
@@ -217,21 +234,34 @@ def get_or_create_sample_sheet(self, flow_cell_name: str) -> None:
         housekeeper or creating it if there is not a valid sample sheet.
         """
         flow_cell: IlluminaRunDirectoryData = self._get_flow_cell(flow_cell_name)
-        LOG.info("Fetching and validating sample sheet from Housekeeper")
+        LOG.info(f"Fetching and validating sample sheet for {flow_cell_name} from Housekeeper")
         try:
             self._use_sample_sheet_from_housekeeper(flow_cell)
             return
-        except SampleSheetError:
+        except SampleSheetContentError:
             LOG.warning(
-                "It was not possible to use sample sheet from Housekeeper, "
-                "trying flow cell sample sheet"
+                f"Validation failed for {flow_cell.get_sample_sheet_path_hk()}. "
+                "Possibly a manually modified sample sheet. Sample sheet will not be re-generated."
+            )
+            return
+        except CgError:
+            LOG.warning(
+                "Sample sheet from Housekeeper is not correctly formatted or does not exist, "
+                "trying sample sheet in sequencing directory"
             )
         try:
             self._use_flow_cell_sample_sheet(flow_cell)
             return
-        except SampleSheetError:
+        except SampleSheetContentError:
+            LOG.warning(
```

Comment:
Why is `SampleSheetContentError` excepted twice in this function? It makes it very hard to understand what's going on

Reply:
We first check the content of the sample sheet in housekeeper, check what kind of error is raising and depending on the type of error we would need to do the same for the sample sheet in the sequencing dir